[BUGFIX] Create CHAR and BINARY as fixed length columns (#5198)
TYPO3-Documentation/TYPO3CMS-Reference-CoreApi: Documentation/ExtensionArchitecture/FileStructure/ExtTablesSql.rst

define only required fields). But the extension manager or admin tools can
handle this.

TYPO3 parses :file:`ext_tables.sql` files into a Doctrine DBAL object schema
to define a virtual database scheme, enriched with
:php-short:`\TYPO3\CMS\Core\Database\SchemaDefaultTcaSchema` information for
`TCA-managed, auto-generated <https://docs.typo3.org/permalink/t3coreapi:auto-generated-db-structure>`_
tables and fields.

Incorrect definitions may not be recognized
by the TYPO3 SQL parser or may lead to SQL errors, when TYPO3 tries
to apply them.

> If TYPO3 is not running on MySQL or a directly compatible
> other :abbr:`DBMS (Database Management System)` like MariaDB, the
> system will parse the file towards the target DBMS like PostgreSQL or SQLite.

Reviewer: ```suggestion
The :file:`ext_tables.sql` file in TYPO3 contains SQL statements written in a TYPO3-specific
format that is not directly valid for any database system. TYPO3 utilizes Doctrine DBAL to
interpret and translate these statements into valid SQL for the specific target :abbr:DBMS (Database Management System), such as MySQL, MariaDB, PostgreSQL, or SQLite.
```

..  _ext-tables-sql-types-char:

Fixed-length SQL type `CHAR`
----------------------------

..  _ext-tables-sql-types-char-varchar-difference:

Key Difference Between CHAR and VARCHAR
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

The main difference between :sql:`CHAR` and :sql:`VARCHAR` is how the database
stores character data in a database. :sql:`CHAR`, which stands for `character`,
is a fixed-length data type, meaning it always reserves a specific amount of
storage space for each value, regardless of whether the actual data occupies
that space entirely. For example, if a column is defined as :sql:`CHAR(10)` and
the word `apple` is stored inside of it, it will still occupy 10 characters worth of
space (not just 5). Unused characters are padded with extra spaces.

On the other hand, :sql:`VARCHAR`, short for `variable character`, is a
variable-length data type. It only uses as much storage space as needed
to store the actual data without padding. So, storing the word `apple` in a
:sql:`VARCHAR(10)` column will only occupy 5 characters worth of
space, leaving the remaining table row space available for other data.

..  _ext-tables-sql-types-char-when:

> When to user `CHAR` columns

Reviewer: ```suggestion
When to use `CHAR` columns
```

~~~~~~~~~~~~~~~~~~~~~~~~~~~

**Rule of thumb for fixed-length** :sql:`CHAR` **columns**

*   Only use with **ensured fixed-length values** (so that no padding occurs).

*   For 255 or more characters :sql:`VARCHAR` or :sql:`TEXT` must be used.

..  _ext-tables-sql-types-char-hints:

> Hints on using fixed-length** `CHAR` columns

Reviewer: ```suggestion
**Hints on using fixed-length** `CHAR` columns
```

Author: Das ist ja schon eine Überschrift, die würde ich nicht noch zusätzlich fe

Reviewer: I don't understand your 2 stars `**` there. I the header should not be bold, please remove the both stars.
